flybywire-aircraft-a320-neo/html_ui/Pages/VCockpit/Instruments/Airliners/FlyByWire_A320_Neo/CDU/ATSU/FansA/A320_Neo_CDU_ATC_LatRequest.js

```diff
@@ -172,8 +172,8 @@ class CDUAtcLatRequest {
         mcdu.onLeftInput[1] = (value) => {
             if (mcdu.currentFlightPhase === FmgcFlightPhases.PREFLIGHT) {
                 // requesting a SID
-                if (mcdu.flightPlanManager.getOrigin() && mcdu.flightPlanManager.getOrigin().ident) {
-                    mcdu.dataManager.GetWaypointsByIdent.bind(mcdu.dataManager)(mcdu.flightPlanManager.getOrigin().ident).then((waypoints) => {
+                if (mcdu.flightPlanManager.getDeparture() && mcdu.flightPlanManager.getDeparture().ident) {
```

Comment:
getDeparture returns the actual SID procedure. The idea is that you don't need to do all this stuff below fetching waypoints etc.

Reply:
I don't get it. The function on the MCDU does not request the entered SID (defined in the flightplan). It is allowed to request any kind of SID. That's why I have to check if the entered SID (in the scratchpad) is valid

Reply:
Ah, I misunderstood what you're doing here :D. Well, I guess we will just have to rewrite the code later.